examples/etl_example.py

```diff
@@ -0,0 +1,169 @@
+    [extract(source_a), extract(source_b), extract(source_c)] (in parallel)
+                           ↓
+                     merge_data
+                           ↓
+                      transform
+                           ↓
+                        load
+
+To run this example:
+
+1. Start Redis (broker and result backend):
+   ```bash
+   redis-server
+   ```
+
+2. Start a Celery worker:
+   ```bash
+   celery -A examples.etl_example worker --loglevel=INFO
+   ```
+
+3. Run the example:
+   ```bash
+   python examples/etl_example.py
+   ```
+
+The pipeline will extract sample data from three sources in parallel,
+merge the results, apply transformations (adding computed fields and filtering),
+and save the final results to a timestamped JSON file.
+"""
+
+from celery import Celery
+from celery import group
+import json
+import random
+from datetime import datetime
+from typing import List, Dict
+
+# Initialize Celery app
+app = Celery('etl_example')
+app.conf.update(
+    broker_url='redis://localhost:6379/0',
+    result_backend='redis://localhost:6379/0'
+)
+
+# Simulated data sources
+DATA_SOURCES = ['source_a', 'source_b', 'source_c']
+
+@app.task(name='etl.extract')
+def extract(source: str) -> List[Dict]:
+    """
+    Extract data from a simulated source.
+    In a real application, this could be a database, API, or file.
+    """
+    # Simulate fetching data with random delay
+    import time
+    time.sleep(random.uniform(0.1, 0.5))
+    
+    # Generate sample data
+    num_records = random.randint(5, 10)
+    records = []
+    
+    for i in range(num_records):
+        record = {
+            'id': f'{source}_{i}',
+            'timestamp': datetime.now().isoformat(),
+            'value': random.randint(1, 100),
+            'source': source
+        }
+        records.append(record)
+    
+    print(f'Extracted {len(records)} records from {source}')
+    return records
+
+@app.task(name='etl.transform')
+def transform(data: List[Dict]) -> List[Dict]:
+    """
+    Transform the extracted data.
+    This example adds some computed fields and filters data.
+    """
+    transformed_data = []
+    
+    for record in data:
+        # Add computed fields
+        transformed_record = record.copy()
+        transformed_record['value_squared'] = record['value'] ** 2
+        transformed_record['is_high_value'] = record['value'] > 50
+        transformed_record['processed_at'] = datetime.now().isoformat()
+        
+        # Only keep records with even values
+        if record['value'] % 2 == 0:
+            transformed_data.append(transformed_record)
+    
+    print(f'Transformed data: kept {len(transformed_data)} out of {len(data)} records')
+    return transformed_data
+
+@app.task(name='etl.merge_data')
+def merge_data(results: List[List[Dict]]) -> List[Dict]:
+    """
+    Merge data from multiple sources into a single list.
+    """
+    merged = []
+    for result in results:
+        merged.extend(result)
```

Comment:
AFAIK list apprehension would be faster here.

Reply:
I'd also suggest something more meaningful here, maybe grouping by date or something more realistic.